jmix-charts/charts-flowui-kit/src/main/java/io/jmix/chartsflowui/kit/component/event/AbstractChartEvent.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.chartsflowui.kit.component.event;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.vaadin.flow.component.ComponentEvent;
+import elemental.json.*;
+import io.jmix.chartsflowui.kit.component.JmixChart;
+import io.jmix.chartsflowui.kit.component.event.dto.BaseChartEventDetail;
+
+
+public abstract class AbstractChartEvent<T extends BaseChartEventDetail> extends ComponentEvent<JmixChart> {
+
+    public static final String EVENT_NAME_PREFIX = "jmixchart";
```

Comment:
Can we rename it to `jmix-chart`? Also, do not forget to rename the corresponding event on the client-side (in the JS component).

Reply:
all DOM events are named without any separators and in lower case(but processed with case incensivity) that may be connected with ability to use autobinded callbacks mapped by property names, so developer can do:

`element.onclick  = () => {}`

instead of: 

`element.addEventListener('click', () => {});`

and this callback will be triggered on event fired. You can check DOM event list in docs, and see that none of them named with special symbols. But this may not work for custom events. I found the advise([https://gomakethings.com/custom-event-naming-conventions-in-vanilla-js](https://gomakethings.com/custom-event-naming-conventions-in-vanilla-js/)) to use library scrope separators as far as word separations. What about to use both '-' and ':' ?